client/webserver/site/src/js/ws.js

```diff
@@ -122,6 +122,7 @@ class MessageSocket {
         if (retrys > 0) {
           retrys = 0
           reloader()
+          return
         }
         forward('open', null, this.handlers)
         while (this.queue.length) {
```

Comment:
So if my read is right, the problem is that we're doing `while (this.queue.length)` here, but for whatever reason (probably a Firefox bug regarding `readyState`), appending to `this.queue` in `this.request`, causing an infinite loop. I think the easiest solution is to do one of these here instead

```
const queue = this.queue
this.queue = []
for (const [route, message] of queue) {
    this.request(route, message)
}
```

Reply:
I guess the real question is if it's 1) just a misfire of the `onopen` handler, or if 2) firefox forgot to set the `readyState`, but is actually ready to send. If 1, we can assume that we'll get another `onopen` when Firefox is ready. If 2, we can't count on another `onopen`, so the queue may never get sent and /markets will hang again.

I'm admittedly having difficulty reproducing, but I have seen it once this morning.

Reply:
The way I've reproduced the error is to:

1. Login to the dex client.
2. Shut down dexc.
3. Click the market selector link. (The spinner starts immediately.)
4. Start dexc. 

I've replaced the while loop with the for loop as you suggested and that does prevent the infinite loop.

Perhaps reloader closes the connection. Then as you pointed out the while loop calls request and the request keeps filling the queue which would keep the while loop active forever as long as readyState remains closed.